diff --git a/vendor/magento/module-customer/Controller/Adminhtml/Index/Save.php b/vendor/magento/module-customer/Controller/Adminhtml/Index/Save.php
index 31971795d25..4589e915f90 100644
--- magento/module-customer/Controller/Adminhtml/Index/Save.php
+++ magento/module-customer/Controller/Adminhtml/Index/Save.php
@@ -27,6 +27,7 @@ use Magento\Customer\Model\CustomerFactory;
 use Magento\Customer\Model\EmailNotificationInterface;
 use Magento\Customer\Model\Metadata\Form;
 use Magento\Customer\Model\Metadata\FormFactory;
+use Magento\Customer\Model\SetCustomerStore;
 use Magento\Framework\Api\DataObjectHelper;
 use Magento\Framework\Api\ExtensibleDataObjectConverter;
 use Magento\Framework\App\Action\HttpPostActionInterface as HttpPostActionInterface;
@@ -75,6 +76,11 @@ class Save extends \Magento\Customer\Controller\Adminhtml\Index implements HttpP
      */
     private $storeManager;

+    /**
+     * @var SetCustomerStore|null
+     */
+    private $customerStore;
+
     /**
      * Constructor
      *
@@ -106,7 +112,9 @@ class Save extends \Magento\Customer\Controller\Adminhtml\Index implements HttpP
      * @param SubscriptionManagerInterface $subscriptionManager
      * @param AddressRegistry|null $addressRegistry
      * @param StoreManagerInterface|null $storeManager
+     * @param SetCustomerStore|null $customerStore
      * @SuppressWarnings(PHPMD.ExcessiveParameterList)
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function __construct(
         Context $context,
@@ -136,7 +144,8 @@ class Save extends \Magento\Customer\Controller\Adminhtml\Index implements HttpP
         JsonFactory $resultJsonFactory,
         SubscriptionManagerInterface $subscriptionManager,
         AddressRegistry $addressRegistry = null,
-        ?StoreManagerInterface $storeManager = null
+        ?StoreManagerInterface $storeManager = null,
+        ?SetCustomerStore $customerStore = null
     ) {
         parent::__construct(
             $context,
@@ -168,6 +177,7 @@ class Save extends \Magento\Customer\Controller\Adminhtml\Index implements HttpP
         $this->subscriptionManager = $subscriptionManager;
         $this->addressRegistry = $addressRegistry ?: ObjectManager::getInstance()->get(AddressRegistry::class);
         $this->storeManager = $storeManager ?? ObjectManager::getInstance()->get(StoreManagerInterface::class);
+        $this->customerStore = $customerStore ?? ObjectManager::getInstance()->get(SetCustomerStore::class);
     }

     /**
@@ -335,6 +345,10 @@ class Save extends \Magento\Customer\Controller\Adminhtml\Index implements HttpP

         if ($this->getRequest()->getPostValue()) {
             try {
+                $this->customerStore->setStore(
+                    $this->getRequest()->getPostValue(CustomerMetadataInterface::ENTITY_TYPE_CUSTOMER)
+                );
+
                 // optional fields might be set in request for future processing by observers in other modules
                 $customerData = $this->_extractCustomerData();

@@ -370,13 +384,6 @@ class Save extends \Magento\Customer\Controller\Adminhtml\Index implements HttpP
                     }
                 }

-                $storeId = $customer->getStoreId();
-                if (empty($storeId)) {
-                    $website = $this->storeManager->getWebsite($customer->getWebsiteId());
-                    $storeId = current($website->getStoreIds());
-                }
-                $this->storeManager->setCurrentStore($storeId);
-
                 // Save customer
                 if ($customerId) {
                     $this->_customerRepository->save($customer);
diff --git a/vendor/magento/module-customer/Controller/Adminhtml/Index/Validate.php b/vendor/magento/module-customer/Controller/Adminhtml/Index/Validate.php
index 41ca037e626..0acc349d65b 100644
--- magento/module-customer/Controller/Adminhtml/Index/Validate.php
+++ magento/module-customer/Controller/Adminhtml/Index/Validate.php
@@ -7,10 +7,12 @@ namespace Magento\Customer\Controller\Adminhtml\Index;

 use Magento\Customer\Api\AccountManagementInterface;
 use Magento\Customer\Api\AddressRepositoryInterface;
+use Magento\Customer\Api\CustomerMetadataInterface;
 use Magento\Customer\Api\CustomerRepositoryInterface;
 use Magento\Customer\Api\Data\AddressInterfaceFactory;
 use Magento\Customer\Api\Data\CustomerInterfaceFactory;
 use Magento\Customer\Model\Address\Mapper;
+use Magento\Customer\Model\SetCustomerStore;
 use Magento\Framework\Api\DataObjectHelper;
 use Magento\Framework\App\Action\HttpGetActionInterface;
 use Magento\Framework\App\Action\HttpPostActionInterface as HttpPostActionInterface;
@@ -33,6 +35,11 @@ class Validate extends CustomerAction implements HttpPostActionInterface, HttpGe
      */
     private $storeManager;

+    /**
+     * @var SetCustomerStore|null
+     */
+    private $customerStore;
+
     /**
      * @param \Magento\Backend\App\Action\Context $context
      * @param \Magento\Framework\Registry $coreRegistry
@@ -60,7 +67,9 @@ class Validate extends CustomerAction implements HttpPostActionInterface, HttpGe
      * @param \Magento\Backend\Model\View\Result\ForwardFactory $resultForwardFactory
      * @param \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory
      * @param StoreManagerInterface|null $storeManager
+     * @param SetCustomerStore|null $customerStore
      * @SuppressWarnings(PHPMD.ExcessiveParameterList)
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function __construct(
         \Magento\Backend\App\Action\Context $context,
@@ -88,7 +97,8 @@ class Validate extends CustomerAction implements HttpPostActionInterface, HttpGe
         \Magento\Framework\View\Result\PageFactory $resultPageFactory,
         \Magento\Backend\Model\View\Result\ForwardFactory $resultForwardFactory,
         \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
-        ?StoreManagerInterface $storeManager = null
+        ?StoreManagerInterface $storeManager = null,
+        ?SetCustomerStore $customerStore = null
     ) {
         parent::__construct(
             $context,
@@ -117,8 +127,8 @@ class Validate extends CustomerAction implements HttpPostActionInterface, HttpGe
             $resultForwardFactory,
             $resultJsonFactory
         );
-
         $this->storeManager = $storeManager ?? ObjectManager::getInstance()->get(StoreManagerInterface::class);
+        $this->customerStore = $customerStore ?? ObjectManager::getInstance()->get(SetCustomerStore::class);
     }

     /**
@@ -145,6 +155,9 @@ class Validate extends CustomerAction implements HttpPostActionInterface, HttpGe
             );
             $customerForm->setInvisibleIgnored(true);

+            $this->customerStore->setStore(
+                $this->getRequest()->getParam(CustomerMetadataInterface::ENTITY_TYPE_CUSTOMER)
+            );
             $data = $customerForm->extractData($this->getRequest(), 'customer');

             if ($customer->getWebsiteId()) {
@@ -161,11 +174,6 @@ class Validate extends CustomerAction implements HttpPostActionInterface, HttpGe
                 $entity_id = $submittedData['entity_id'];
                 $customer->setId($entity_id);
             }
-            if (isset($data['website_id']) && is_numeric($data['website_id'])) {
-                $website = $this->storeManager->getWebsite($data['website_id']);
-                $storeId = current($website->getStoreIds());
-                $this->storeManager->setCurrentStore($storeId);
-            }
             $errors = $this->customerAccountManagement->validate($customer)->getMessages();
         } catch (\Magento\Framework\Validator\Exception $exception) {
             /* @var $error Error */
